src/modules/Crowbar.rb

```diff
@@ -118,6 +93,13 @@ def json2hash(file_name)
       ret
     end
 
+    # read given yaml file and return the content as a map
+    def yaml2hash(file_name)
+      ret = YAML.load(File.read(file_name))
```

Comment:
re option 3: it should probably be in InitRepoURL, not yet sure if in other places as well